### ISSUE TITLE
Chef_upgrade task stuck on (Running) doesn't complete upgrade on Windows

### ISSUE THREAD

**Issue by iasllani** [chef-cookbooks/chef_client_updater#172]
### Cookbook version
[Version of the cookbook where you are encountering the issue]
3.8.0

### Chef-client version
[Version of chef-client in your environment]
12.9.41

### Platform Details
[Operating system distribution and release version. Cloud provider if running in the cloud]
Windows 2008 R2

### Scenario:
[What you are trying to achieve and you can't?]
Trying to upgrade client from 12.9.41, to 15.2.20

### Steps to Reproduce:
[If you are filing an issue what are the things we need to do in order to repro your problem? How are you using this cookbook or any resources it includes?]
Run the chef_client_updater cookbook version 3.8.0, changed post action to both exec and kill and got same result.
**THIS DID NOT HAPPEN IN PREVIOUS RELEASE**

### Expected Result:
[What are you expecting to happen as the consequence of above reproduction steps?]
Chef-client force quits, chef_upgrade runs, removes 12.9.41 then new client 15.2.20 is now running

### Actual Result:
[What actually happens after the reproduction steps? Include the error output or a link to a gist if possible.]
I can see the chef_upgrade task running, however, it seems to never stop- Windows shows that client version is still 12.9.41, however, it needs to be repaired. Again on 3.7.3 the client would be update successfully. (However EULA license attribute did not work)

![image](https://user-images.githubusercontent.com/38671335/71362979-a3e77100-2565-11ea-8dbc-c57a32c0f25f.png)

**Comment by iasllani** [comment]
any updates?

**Comment by cdbeard2016** [comment]
Hi Iasllani,

I ran into this today on 2012r2. The scheduled task launches a powershell script which the cookbook writes to c:\opscode\chef\chef_upgrade.ps1 by default. In my case I got two interactive prompts for maxCount and inMinutes which I suspect might be preventing the task from proceeding. If you run that script in an admin powershell session what does it give you?

I had a poke around in the providers/default.rb and found the following:
```
def wait_for_chef_client_or_reschedule_upgrade_task_function
  <<-EOH
  Function WaitForChefClientOrRescheduleUpgradeTask {
    <# Wait for running chef-client to finish up to n times.  If it has not finished after maxcount tries, then reschedule the upgrade task inMinutes minutes in the future and exit.
    #>
    param(
          [Parameter(Mandatory=$true)]
          [Int]$maxcount = 5,
          [Parameter(Mandatory=$true)]
          [Int]$inMinutes = 10
    )
```

I cloned a copy of the cookbook and changed both $true to $false and it worked again for me on a clean run. It looks like this was changed in 2441ae6, so @gholtiii is there a reason why these parameters are mandatory? I think in most cases the script isn't intended to be run interactively so I think we're fairly safe to take those constraints off?

Kind Regards,

Chris

**Comment by iasllani** [comment]
great find. tested on 2008r2 and the client is upgraded, still per https://github.com/chef-cookbooks/chef_client_updater/issues/172 license is not accepted despite defining attribute with either 'accept' or 'accept-no-persist'

**Comment by tas50** [comment]
Fixed in 3.8.2
